### PR TITLE
Document toolchain for Groovy compile

### PR DESCRIPTION
**Pull request by ljacomet**
Also includes DSL level documentation for the toolchain feature in general.

### CODE REVIEW

**Comment by ljacomet** [comment]
@bot-gradle test RFM

File: subprojects/docs/src/docs/userguide/jvm/groovy_plugin.adoc [hunk 1]
```diff
@@ -202,7 +202,10 @@ The Groovy plugin also modifies some source set properties:
 [[sec:groovyCompile]]
 == GroovyCompile
 
-The Groovy plugin adds a link:{groovyDslPath}/org.gradle.api.tasks.compile.GroovyCompile.html[GroovyCompile] task for each source set in the project. The task type extends the `JavaCompile` task (see <<building_java_projects.adoc#sec:compile, the relevant Java Plugin section>>). The `GroovyCompile` task supports most configuration options of the official Groovy compiler.
+The Groovy plugin adds a link:{groovyDslPath}/org.gradle.api.tasks.compile.GroovyCompile.html[GroovyCompile] task for each source set in the project.
+The task type extends the `JavaCompile` task (see <<building_java_projects.adoc#sec:compile, the relevant Java Plugin section>>).
```
Comment (bmuskalla) [inline subprojects/docs/src/docs/userguide/jvm/groovy_plugin.adoc:206]:
```suggestion
The task type extends the `AbstractCompile` task (see <<building_java_projects.adoc#sec:compile, the relevant Java Plugin section>>).
```

File: subprojects/plugins/src/main/java/org/gradle/api/plugins/JavaPluginExtension.java [hunk 2]
```diff
@@ -129,17 +129,19 @@
     ModularitySpec getModularity();
 
     /**
-     * Configure the toolchain requirements for tasks that require a tool from the toolchain (e.g. {@link org.gradle.api.tasks.compile.JavaCompile}.
+     * Gets the project wide toolchain requirements that will be used for tasks requiring a tool from the toolchain (e.g. {@link org.gradle.api.tasks.compile.JavaCompile}).
+     * <p>
+     * Configuring a toolchain makes it illegal to set the {@code sourceCompatibility} or {@code targetCompatibility} on this extension.
```
Comment (bmuskalla) [inline subprojects/plugins/src/main/java/org/gradle/api/plugins/JavaPluginExtension.java:134]:
```suggestion
     * Configuring a toolchain cannot be used together with {@code sourceCompatibility} or {@code targetCompatibility} on this extension. Both values will be enforced by the toolchain.
```

**Review by bmuskalla — CHANGES_REQUESTED** [review]
LGTM. You missed to fix `toolchains.adoc:47` though.

File: subprojects/docs/src/docs/userguide/jvm/groovy_plugin.adoc (hunk 1 above)
Comment (ljacomet) [reply subprojects/docs/src/docs/userguide/jvm/groovy_plugin.adoc:206]: Thanks for that, did a slightly different fix in the end.

File: subprojects/plugins/src/main/java/org/gradle/api/plugins/JavaPluginExtension.java (hunk 2 above)
Comment (ljacomet) [reply subprojects/plugins/src/main/java/org/gradle/api/plugins/JavaPluginExtension.java:134]: Replaced enforced with sourced in your suggestion.

**Comment by ljacomet** [comment]
Thanks for the reminder to fix the toolchains doc itself, added.

**Comment by ljacomet** [comment]
@bot-gradle test RFM

**Pull request merged**